src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -463,6 +464,94 @@ namespace Abs {
 
 } // namespace Abs
 
+namespace Ord{
+    static ASR::expr_t* eval_Ord(Allocator &al, const Location &loc,
+            Vec<ASR::expr_t*> &args) {
+        LCOMPILERS_ASSERT(ASRUtils::all_args_evaluated(args));
+        ASR::ttype_t *type = ASRUtils::TYPE(ASR::make_Integer_t(al, loc, 4, nullptr, 0));
+        ASR::expr_t *expr = args[0];
+        ASR::ttype_t *t = ASRUtils::expr_type(expr);
+        if (ASRUtils::is_character(*t)) {
+            int64_t val;
+            if (ASRUtils::expr_value(expr)) {
+                char* c = ASR::down_cast<ASR::StringConstant_t>(expr)->m_s;
+                val = static_cast<int64_t>(c[0]);
+            }
+            return ASRUtils::EXPR(ASR::make_IntegerConstant_t(
+               al, loc, val, type));
+        } else {
+            return nullptr;
+        }
+    }
+    static inline ASR::asr_t* create_Ord(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() != 1) {
+            err("Intrinsic ord function accepts exactly 1 argument", loc);
+        }
+        ASR::ttype_t *type = ASRUtils::expr_type(args[0]);
+        if (ASRUtils::is_character(*type)) {
+            char* c = ASR::down_cast<ASR::StringConstant_t>(ASRUtils::expr_value(args[0]))->m_s;
+            if ((ASRUtils::expr_value(args[0]) != nullptr) && std::string(c).size() != 1) {
+                    err("ord() is only supported for String of length 1", args[0]->base.loc);
+                }
+        }
```

Comment:
Here don't check for character type instead check for StringConstant and length of the character
For `StringConstant` the `expr_value` returns `StringConstant` itself, so instead check that args[0] is a StringConstant.

Also, as this PR only compute value for compile-time value, throw error for runtime input.

Reply:
You can check, how partition is handled in this PR: https://github.com/lcompilers/lpython/pull/1673